### PR TITLE
Introducing auto flush in batcherImpl

### PR DESCRIPTION
**Pull request by rahulKQL**

### CODE REVIEW

File: gax/src/main/java/com/google/api/gax/batching/v2/BatcherImpl.java [hunk 1]
```diff
@@ -74,27 +87,32 @@ public BatcherImpl(
...
   public ApiFuture<ElementResultT> add(ElementT element) {
     Preconditions.checkState(!isClosed, "Cannot add elements on a closed batcher");
+    SettableApiFuture<ElementResultT> result = SettableApiFuture.create();
 
     if (currentOpenBatch == null) {
       currentOpenBatch = new Batch<>(prototype, batchingDescriptor, batchingSettings);
+      currentOpenBatch.add(element, result);
+      executor.schedule(pushCurrentBatchRunnable, maxDelay.toMillis(), TimeUnit.MILLISECONDS);
+    } else {
+      currentOpenBatch.add(element, result);
```
Comment (rahulKQL) [inline gax/src/main/java/com/google/api/gax/batching/v2/BatcherImpl.java]: How else can I refactor it to reduce this duplicate code... all suggestions are welcome.
(some background info: `executor.schedule(pushCurrentBatchRunnable, maxDelay...` can auto trigger it and can make `currentOpenBatch` `null`).

File: gax/src/main/java/com/google/api/gax/batching/v2/BatchingSettings.java [hunk 2]
```diff
@@ -32,17 +32,18 @@
...
  * Represents the batching settings to use for an API method that is capable of batching.
  *
  * <p>Each batching client must define their own set of default values for these thresholds, which
  * would be the safest behavior for their jobs.
  *
- * <p>The default instance of this settings are configured to accept elements until either of the
- * threshold reaches {@link Long#MAX_VALUE}) or an explicit call to {@link Batcher#flush()} is made
- * or the {@link Batcher#close()} is called. Users are expected to configure actual batching
- * thresholds explicitly: the element count or the request bytes count.
+ * <p>The default instance of this settings are configured to accept elements until any of the
```
Comment (romlogic) [inline gax/src/main/java/com/google/api/gax/batching/v2/BatchingSettings.java]: these settings are configured to ...

File: gax/src/main/java/com/google/api/gax/batching/v2/BatchingSettings.java [hunk 3]
```diff
@@ -56,12 +57,14 @@
  * <p>There are several supported thresholds:
  *
  * <ul>
+ *   <li><b>Delay Threshold</b>: Counting from the time when the first message is queued, once this
+ *       duration has passed, then send the batch. The default value is 1 minutes.
```
Comment (romlogic) [inline gax/src/main/java/com/google/api/gax/batching/v2/BatchingSettings.java]: 1 minute.

File: gax/src/main/java/com/google/api/gax/batching/v2/BatchingSettings.java (hunk 2 above)
Comment (rahulKQL) [reply gax/src/main/java/com/google/api/gax/batching/v2/BatchingSettings.java]: "instance of this" tells about a single instance of `BatchingSettings.java`

**Pull request closed without merging**